### PR TITLE
User model with devise

### PR DESCRIPTION
**Pull request by pospieszynski**
@gotar 
Basic authentication setup with Devise gem. I have overwritten Registration controller to be able to use custom fields: name and surname when registering a user. I have also added input fields in sign_up view.

### CODE REVIEW

**Comment by taralalukasz** [comment]
I'm not sure if comments are necessary in some of the files there, maybe it would be better to remove them ?

**Comment by crappylime** [comment]
If you don't have anything in spec file, maybe you don't need it, on the other hand you can add some validations in model and test them

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class RegistrationsController < Devise::RegistrationsController
+
```
Comment (neonowy) [inline app/controllers/registrations_controller.rb]: This empty line is not needed imho.

File: config/database.yml [hunk 2]
```diff
@@ -17,7 +17,6 @@ default: &default
   # For details on connection pooling, see rails configuration guide
   # http://guides.rubyonrails.org/configuring.html#database-pooling
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-
```
Comment (neonowy) [inline config/database.yml]: Why did you delete this line?

Comment (neonowy) [inline config/database.yml]: Why did you delete this line?

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context "With a proper validations setup user" do
+    before :each do
+      @user = User.create(name: "Wojtek", surname: "Pośpiech")
+    end
+    it "should not be valid without a name" do
+      @user.name = nil
+      @user.should_not be_valid
```
Comment (tprochniewicz) [inline spec/models/user_spec.rb]: Don't use should, use expect

**Comment by filip373** [comment]
:+1:

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context "With a proper validations setup user" do
+    let(:user) { User.create(name: "Wojtek", surname: "Pośpiech", email: "a@a.pl", password: "topsecret") }
+    it "should not be valid without a name" do
+      user.name = nil
+      expect(user).not_to be_valid
+    end
+    it "should not be valid without a surname" do
```
Comment (tprochniewicz) [inline spec/models/user_spec.rb]: Add new context for not valid params

Comment (gotar) [inline spec/models/user_spec.rb]: do not use should world in test description

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  context "With a proper validations setup user" do
+    let(:user) { User.create(name: "Wojtek", surname: "Pośpiech", email: "a@a.pl", password: "topsecret") }
+    it "should not be valid without a name" do
```
Comment (gotar) [inline spec/models/user_spec.rb]: add empty lines

it looks like first 2 weeks teach nothing 😢

Comment (tprochniewicz) [inline spec/models/user_spec.rb]: it was still in progress, not marked to checked

**Comment by taralalukasz** [comment]
+1

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  context "With improper validations setup user" do
+    it "should not be valid without a name" do
+      user.name = nil
+      expect(user).not_to be_valid
+    end
+
+    it "should not be valid without a surname" do
+      user.surname = nil
+      expect(user).not_to be_valid
+    end
```
Comment (gotar) [inline spec/models/user_spec.rb]: drop should from test description

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -9,6 +9,8 @@
   </head>
 
   <body>
+    <p class="notice"><%= notice %></p>
+    <p class="alert"><%= alert %></p>
```
Comment (gotar) [inline app/views/layouts/application.html.erb]: can be changed to flash helper http://stackoverflow.com/questions/12345065/rails-flash-message-helper

File: app/views/devise/registrations/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    <%= f.label :password_confirmation %><br/>
+    <%= f.password_field :password_confirmation, autocomplete: "off" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (gotar) [inline app/views/devise/registrations/new.html.erb]: generic devise views? if yes drop them

Comment (pospieszynski) [inline app/views/devise/registrations/new.html.erb]: no, these are customized views for the application purposes(additional name, surname fields)

File: spec/models/user_spec.rb (hunk 6 above)
Comment (pospieszynski) [inline spec/models/user_spec.rb]: done
